3684/0/about_proxy_object_project.rb

```diff
@@ -13,28 +13,57 @@
 # of the Proxy class is given in the AboutProxyObjectProject koan.
 
 class Proxy
-  attr_reader :messages
-
+  attr_reader :object
   def initialize(target_object)
     @object = target_object
-    @messages = []
+    @messages_array = []
+  end
+
+  def power
+    @object.power
+    @messages_array << :power
+  end
+
+  def channel=(value)
+    @object.channel = value
+    @messages_array << :channel=
+  end
+
+  def channel
+    @object.channel
   end
 
-  def respond_to_missing?(name, include_private); end
+  def on?
+    @object.on?
+    @messages_array << :on?
+  end
+
+  def number_of_times_called(parameter)
+    @messages_array.count(parameter)
+  end
 
-  def method_missing(sym, *args, &block)
-    @messages << sym
-    return @object.send(sym, *args, &block)
-    super # rubocop:disable Lint/UnreachableCode
+  def messages
+    @messages_array
   end
 
-  def called?(method)
-    @messages.include?(method)
+  def called?(parameter)
+    if @messages_array.include?(parameter)
+      true
+    else
+      false
+    end
+  end
```

Comment:
please, play with .include?(). maybe in the `irb` console. your changes made things worse.

Reply:
done